### ISSUE TITLE
`terraform show` crashes when `AWS_PROFILE` env is missing

### ISSUE THREAD

**Issue by sateeshblue** [hashicorp/terraform#31127]
### Terraform Version

```
Terraform v1.2.1
on linux_amd64
+ provider registry.terraform.io/cloudposse/awsutils v0.11.0
+ provider registry.terraform.io/hashicorp/aws v4.11.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/random v3.1.3
+ provider registry.terraform.io/hashicorp/tls v3.3.0
```

### Terraform Configuration Files

```terraform
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

can't construct diagnostic(s) from string
goroutine 1 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x19
github.com/hashicorp/terraform/internal/logging.PanicHandler()
        /home/circleci/project/project/internal/logging/panic.go:55 +0x153
panic({0x22b6a40, 0xc000d437c0})
        /usr/local/go/src/runtime/panic.go:844 +0x258
github.com/hashicorp/terraform/internal/tfdiags.Diagnostics.Append({0x0?, 0xb8?, 0x26b428d?}, {0xc000c1f7f0?, 0x1, 0x3e8a710?})
        /home/circleci/project/project/internal/tfdiags/diagnostics.go:86 +0xa0e
github.com/hashicorp/terraform/internal/command.(*ShowCommand).showFromLatestStateSnapshot(0xc000188000)
        /home/circleci/project/project/internal/command/show.go:154 +0x245
github.com/hashicorp/terraform/internal/command.(*ShowCommand).show(0xc000188000, {0x0, 0x0})
        /home/circleci/project/project/internal/command/show.go:93 +0x6d
github.com/hashicorp/terraform/internal/command.(*ShowCommand).Run(0xc000188000, {0xc00004c050?, 0xffffffffffffffff?, 0x0?})
        /home/circleci/project/project/internal/command/show.go:51 +0x218
github.com/mitchellh/cli.(*CLI).Run(0xc0002c6500)
        /home/circleci/go/pkg/mod/github.com/mitchellh/cli@v1.1.3/cli.go:262 +0x5f8
main.realMain()
        /home/circleci/project/project/main.go:312 +0x15b4
main.main()
        /home/circleci/project/project/main.go:58 +0x19
```

### Expected Behavior

terraform listing

### Steps to Reproduce

`terraform show`

### Additional Context

exporting the `AWS_PROFILE` fixes the issue

**Comment by crw** [comment]
Thanks for the report!

**Comment by bschaatsbergen** [comment]
I'll happily look into this.

**Comment by richardkmichael** [comment]
I have the same crash, but it's not with a missing `AWS_PROFILE` -- instead it happens if the backend is unavailable.

Terraform 1.2.4, configured to use the HTTP backend, stored in a GitLab repo.  Without a username and password (`TF_HTTP_USERNAME|PASSWORD`), it crashes.

```
terraform {
  backend "http" {
    address = "https://gitlab.com/api/v4/projects/NNN/terraform/state/default"
    lock_address = "https://gitlab.com/api/v4/projects/NNN/terraform/state/default/lock"
    unlock_address = "https://gitlab.com/api/v4/projects/NNN/terraform/state/default/lock"
    lock_method = "POST"
    unlock_method = "DELETE"
    retry_wait_min = 5
  }
}
```

```
 $ terraform show
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

can't construct diagnostic(s) from string
goroutine 1 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x68
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x20
github.com/hashicorp/terraform/internal/logging.PanicHandler()
        /Users/distiller/project/project/internal/logging/panic.go:55 +0x174
panic({0x1068b2020, 0x14000b928b0})
        /usr/local/go/src/runtime/panic.go:844 +0x258
github.com/hashicorp/terraform/internal/tfdiags.Diagnostics.Append({0x0?, 0x3e?, 0x1060bc6be?}, {0x140009bf7b8?, 0x1, 0x16?})
        /Users/distiller/project/project/internal/tfdiags/diagnostics.go:86 +0x8b4
github.com/hashicorp/terraform/internal/command.(*ShowCommand).showFromLatestStateSnapshot(0x14000603860)
        /Users/distiller/project/project/internal/command/show.go:154 +0x1c8
github.com/hashicorp/terraform/internal/command.(*ShowCommand).show(0x14000603860, {0x0, 0x0})
        /Users/distiller/project/project/internal/command/show.go:93 +0x50
github.com/hashicorp/terraform/internal/command.(*ShowCommand).Run(0x14000603860, {0x14000110030?, 0x6?, 0xffffffffffffffff?})
        /Users/distiller/project/project/internal/command/show.go:51 +0x1a8
github.com/mitchellh/cli.(*CLI).Run(0x1400087eb40)
        /Users/distiller/go/pkg/mod/github.com/mitchellh/cli@v1.1.4/cli.go:262 +0x4cc
main.realMain()
        /Users/distiller/project/project/main.go:312 +0x12b0
main.main()
        /Users/distiller/project/project/main.go:58 +0x20

 $ terraform -version
Terraform v1.2.4
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.14.0
+ provider registry.terraform.io/hashicorp/random v3.1.0

$ uname -a
Darwin air.local 20.6.0 Darwin Kernel Version 20.6.0: Tue Apr 19 21:04:40 PDT 2022; root:xnu-7195.141.29~1/RELEASE_ARM64_T8101 arm64
```

**Comment by jbardin** [comment]
Closed via #31444
